Sources/MapboxNavigation/OverlayController.swift

```diff
@@ -0,0 +1,390 @@
+     */
+    private func tileSetIdentifiers(_ sourceIdentifier: String, sourceType: String) -> [String] {
+        do {
+            if sourceType == "vector",
+               let properties = try navigationMapView.mapView.__map.getStyleSourceProperties(forSourceId: sourceIdentifier).value as? Dictionary<String, Any>,
+               let url = properties["url"] as? String,
+               let configurationURL = URL(string: url),
+               configurationURL.scheme == "mapbox",
+               let tileSetIdentifiers = configurationURL.host?.components(separatedBy: ",") {
+                return tileSetIdentifiers
+            }
+        } catch {
+            NSLog("Failed to get source properties with error: \(error.localizedDescription).")
+        }
+        
+        return []
+    }
+    
+    private func roadShieldTextColor(line: Feature) -> UIColor? {
+        guard let shield = line.properties?["shield"] as? String else {
+            return nil
+        }
+        
+        // shield_text_color is present in Mapbox Streets source v8 but not v7.
+        guard let shieldTextColor = line.properties?["shield_text_color"] as? String else {
+            let currentShield = HighwayShield.RoadType(rawValue: shield)
+            return currentShield?.textColor
+        }
+        
+        switch shieldTextColor {
+        case "black":
+            return .black
+        case "blue":
+            return .blue
+        case "white":
+            return .white
+        case "yellow":
+            return .yellow
+        case "orange":
+            return .orange
+        default:
+            return .black
+        }
+    }
+    
+    private func roadShieldAttributedText(for text: String, textColor: UIColor, imageName: String) -> NSAttributedString? {
+        guard let image = navigationMapView.mapView.style.getStyleImage(with: imageName)?.cgImage() else { return nil }
+        let attachment = ShieldAttachment()
+        attachment.image = UIImage(cgImage: image.takeRetainedValue()).withCenteredText(text,
+                                                                                        color: textColor,
+                                                                                        font: UIFont.boldSystemFont(ofSize: UIFont.systemFontSize),
+                                                                                        scale: UIScreen.main.scale)
+        return NSAttributedString(attachment: attachment)
+    }
+    
+    /**
+     Method, which returns a boolean value indicating whether the tile source is a supported version of the Mapbox Streets source.
+     */
+    private func isMapboxStreets(_ identifiers: [String]) -> Bool {
+        return identifiers.contains("mapbox.mapbox-streets-v8") || identifiers.contains("mapbox.mapbox-streets-v7")
+    }
```

Comment:
This method should be a static method of an extension of VectorSource. It was originally on an extension of MGLVectorTileSource but got inlined in RouteMapViewController when upgrading to map SDK v10.

Reply:
Done